io_scene_nif/modules/nif_import/property/geometry/mesh.py

```diff
@@ -48,35 +51,56 @@
 class MeshPropertyProcessor:
 
     def __init__(self):
-        self.niproperty = NiPropertyProcessor.get()
-        self.bsshader = BSShaderPropertyProcessor.get()
-        self.bsshaderlighting = BSShaderLightingPropertyProcessor.get()
+        # get processor singletons
+        self.processors = (
+            NiPropertyProcessor.get(),
+            BSShaderPropertyProcessor.get(),
+            BSShaderLightingPropertyProcessor.get()
+        )
 
         # Register processors
         self.process_property = singledispatch(self.process_property)
-        self.niproperty.register_niproperty(self.process_property)
-        self.bsshader.register_bsproperty(self.process_property)
-        self.bsshaderlighting.register_bsproperty(self.process_property)
+        for processor in self.processors:
+            processor.register(self.process_property)
 
     def process_property_list(self, n_block, b_mesh):
-        self.niproperty.b_mesh = b_mesh
-        self.niproperty.n_block = n_block
-        self.bsshader.b_mesh = b_mesh
-        self.bsshader.n_block = n_block
-        self.bsshaderlighting.b_mesh = b_mesh
-        self.bsshaderlighting.n_block = n_block
+        # get all valid properties that are attached to n_block
+        props = list(prop for prop in itertools.chain(n_block.properties, n_block.bs_properties) if prop is not None)
+        # just to avoid duped materials, a first pass, make sure a named material is created
+        for prop in props:
+            if prop.name:
+                print("Found name for mat")
+                name = prop.name.decode()
+                if name and name in bpy.data.materials:
+                    b_mat = bpy.data.materials[name]
+                    # todo [material] fixme - we have to avoid multiple passes on the same material
+                    # it seems to mess with the singleton, or the bmat
+                    b_mesh.materials.append(b_mat)
+                    NifLog.debug(f"Retrieved already imported material {b_mat} from name {name}")
```

Comment:
`Retrieved already imported material` > `Duplicate material {b_mat} from name {name} already found`

Reply:
Also like the f-string formatting.